### PR TITLE
Provide option to fine tune spacy NER model instead of training from scratch

### LINKED ISSUE #136: Fixes for NER training with spaCy
tl;dr: Consider removing `nlp.entity.end_training()` in spaCy NER training.

Details here: https://github.com/explosion/spaCy/issues/773

Have been working hard on the neural network model, to be launched with spaCy 2.0. This should give much improved performance for both NER and intent classification.

### LINKED ISSUE #166: Spacy Built-in entity types
Hi,
Can we use Spacy Built-in entity types in rasa_nlu? https://spacy.io/docs/usage/entity-recognition#entity-types
Or we need to use them directly from Spacy?

### PR DESCRIPTION
**Pull request by tmbo**
first commit:
- fixed regression of #152 (trainer `persist` got screwed up & entity extractor was not set after training)

second commit:
- improved training tests (they now check if entity extractor & intent classifier exist)
- fixes #166 (allows to fine tune spacys provided NER models with the training data)
- don't `finish` after entity training. fixes #136

### CODE REVIEW

**Comment by tmbo** [comment]
Documentation will be added if approved.

**Comment by tmbo** [comment]
- note to myself: add hint to docu that spacy needs a lot of training data for NER

File: _pytest/test_train.py [hunk 1]
```diff
@@ -25,7 +23,9 @@ def test_train_mitie():
         "path": tempfile.mkdtemp(),
         "data": "./data/examples/rasa/demo-rasa.json"
     }
-    run_train(_config)
+    trained = run_train(_config)
+    assert trained.entity_extractor is not None
+    assert trained.intent_classifier is not None
 
 
 def test_train_mitie_sklearn():
```
Comment (amn41) [inline _pytest/test_train.py]: why no asserts at the end of this train?

**Comment by amn41** [comment]
Looks good. Please go ahead and add docs. I made one small comment (think you forgot a couple of `assert`s in `test_train.py`

File: _pytest/test_train.py (hunk 1 above)
Comment (tmbo) [inline _pytest/test_train.py]: good catch, merge issue...

**Pull request merged**